embl-api-core/src/main/java/uk/ac/ebi/embl/api/validation/check/file/SubmissionFile.java

```diff
@@ -1,25 +1,46 @@
 package uk.ac.ebi.embl.api.validation.check.file;
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
 
-public enum SubmissionFile {
+public class SubmissionFile {
 
-	FASTA, FLATFILE;
-	List<File> files = new ArrayList<File>();
+	public enum FileType {
+		FASTA,
+		FLATFILE,
+		AGP,
+		CHROMOSOME_LIST,
+		UNLOCALISED_LIST
+	};
 
-	public void addFile(File file) {
-		files.add(file);
+	private final FileType fileType;
+	private final File file;
+	private final File fixedFile;
+
+	public SubmissionFile (FileType fileType, File file) {
+		this.fileType = fileType;
+		this.file = file;
+		this.fixedFile = null;

```

Comment:
where do we use fixedFile field in submissionfile class

Reply:
Iain generates fixed files during validation in the transcriptome and template pipelines. Later in the pipelines, these files are read and accessions are injected. Finally, these entries are written to immutable file products and also loaded into ENAPRO.

Reply:
i think, we can set this property to submissionvalidationplanproperty